### DISCUSSION TITLE
How to expand all rows by default?

### DISCUSSION THREAD

**Discussion by Kavimugilan17** [TanStack/table#2134] [General] [answered]
HI,
 How to expand all rows by default. 
Can you please anyone help?

Codesandbox Link: 
https://codesandbox.io/s/competent-cohen-bsnw3?file=/src/App.js

**Comment by rantibi** [comment]
just add the following:
`  useEffect(() => {
    toggleAllRowsExpanded();
  }, []);`

**Reply by Kavimugilan17** [reply]
Great!!, Thank you so much. It's working fine.

**Reply by KevinVandy** [reply]
I don't know why this is the answer. Bad practice here

**Comment by dclark27** [comment]
Please don't use an effect to achieve this. Simply set expanded to true when creating your useState for the expanding.

```ts
const [expanded, setExpanded] = useState<ExpandedState>(true);
```

**Comment by brmzkw** [comment, marked as answer]
If tanstack table controls the state:

```ts
  const table = useReactTable({
   …
    initialState: { expanded: true },
    getExpandedRowModel: getExpandedRowModel(),
  });
```
